cpu/sources/layout7/cpu_oods.move

```diff
@@ -0,0 +1,940 @@
+                res =
+                    res +
+                    fmul(fmul(/*(x - z^2)^(-1)*/ *borrow(batch_inverse_array, *denominators_ptr + 98),
+                        oods_alpha_pow),
+                        column_value + (prime -/*composition_oods_values[0]*/ *borrow(ctx, 359 + 192)))
+                ;
+                oods_alpha_pow = fmul(oods_alpha_pow, oods_alpha);
+            };
+
+            {
+                // Read the next element.
+                let column_value = fmul(*borrow(ctx, *composition_query_responses + 1), K_MONTGOMERY_R_INV);
+                // res += c_193*(h_1(x) - C_1(z^2)) / (x - z^2).
+                res =
+                    res +
+                    fmul(fmul(/*(x - z^2)^(-1)*/ *borrow(batch_inverse_array, *denominators_ptr + 98),
+                        oods_alpha_pow),
+                        column_value + (prime - /*composition_oods_values[1]*/ *borrow(ctx, 359 + 193)))
+                ;
+                oods_alpha_pow = fmul(oods_alpha_pow, oods_alpha);
+            };
+
+            // Advance composition_query_responses by amount read (0x20 * constraintDegree).
+            *composition_query_responses = *composition_query_responses + 2;
+
+            // Append the friValue, which is the sum of the out-of-domain-sampling boundary
+            // constraints for the trace and composition polynomials, to the fri_queue array.
+            set_el(ctx, *fri_queue + 1, res % prime);
+
+            // print(&(res % prime));
+
+            // Append the friInvPoint of the current query to the fri_queue array.
+            set_el(ctx, *fri_queue + 2, *borrow(batch_inverse_array, *denominators_ptr + 99));
+
+            // Advance denominators_ptr by chunk size (0x20 * (2+N_ROWS_IN_MASK)).
+            *denominators_ptr = *denominators_ptr + 100;
+
+            *fri_queue = *fri_queue + 3;
+        };
+
+        if (*fri_queue >= *fri_queue_end) {
+            *first_invoking = true;
+            *checkpoint = FB_CHECKPOINT1;
+            true
+        } else {
+            false
+        }
+    }
+
+    /*
+          Computes and performs batch inverse on all the denominators required for the out of domain
+          sampling boundary constraints.
+
+          Since the frieval_points are calculated during the computation of the denominators
+          this function also adds those to the batch inverse in prepartion for the fri that follows.
+
+          After this function returns, the batch_inverse_out array holds #queries
+          chunks of size (2 + N_ROWS_IN_MASK) with the following structure:
+          0..(N_ROWS_IN_MASK-1):   [(x - g^i * z)^(-1) for i in rowsInMask]
+          N_ROWS_IN_MASK:          (x - z^constraintDegree)^-1
+          N_ROWS_IN_MASK+1:        frieval_pointInv.
+    */
+    fun oods_prepare_inverses(ctx: &vector<u256>, batch_inverse_array: &mut vector<u256>) {
+        let eval_coset_offset_ = GENERATOR_VAL;
```

Comment:
```suggestion
        let eval_coset_offset = GENERATOR_VAL;
```

Reply:
??????

Reply:
it was named as such in the original contract, is it a convention? :))